### PR TITLE
use primitive/container terminology outside zcode

### PR DESCRIPTION
**Pull request by nwt**
Follow-on to #197.

### CODE REVIEW

File: zbuf/record.go [hunk 1]
```diff
@@ -205,8 +205,8 @@ func (r *Record) TypeCheck() error {
 var (
 	ErrMissingField = errors.New("record missing a field")
 	ErrExtraField   = errors.New("record with extra field")
-	ErrNotContainer = errors.New("scalar where container was expected")
-	ErrNotScalar    = errors.New("container where scalar was expected")
+	ErrNotContainer = errors.New("primitive where container was expected")
+	ErrNotPrimitive = errors.New("container where primitive was expected")
```
Comment (aswan) [inline zbuf/record.go]: nit: these sound awkward to me.  Is "got (or saw? found?) primitive type where container was expected" (and vice versa) too wordy?

Comment (henridf) [reply zbuf/record.go]: i think i came up with these, if the proposed replacement is less awkward then sounds good to me.
